src/main/java/org/sonar/plugins/csharp/sarif/SarifParser01And04.java

```diff
@@ -17,59 +17,48 @@
  * along with this program; if not, write to the Free Software Foundation,
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
-package org.sonar.plugins.csharp;
+package org.sonar.plugins.csharp.sarif;
 
-import com.google.common.base.Charsets;
-import com.google.common.io.Files;
 import com.google.gson.JsonArray;
 import com.google.gson.JsonElement;
 import com.google.gson.JsonObject;
 import com.google.gson.JsonParser;
-import java.io.File;
-import java.io.IOException;
 
-public class SarifParser {
+public class SarifParser01And04 implements SarifParser {
 
   private static final String FILE_PROTOCOL = "file:///";
-  private SarifParserCallback callback;
+  private String contents;
 
-  public SarifParser(SarifParserCallback callback) {
-    this.callback = callback;
+  SarifParser01And04(String contents) {
+    this.contents = contents;
   }
 
-  public void parse(File file) {
-    String contents;
-    try {
-      contents = Files.toString(file, Charsets.UTF_8);
-    } catch (IOException e) {
-      throw new IllegalStateException("Unable to read the Roslyn SARIF report file: " + file.getAbsolutePath(), e);
-    }
-
+  public void parse(SarifParserCallback callback) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add the "@Override" annotation above this method signature [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1161)